app/models/pipeline_run_stage.rb

```diff
@@ -118,8 +122,17 @@ def redacted_dag_json
   def run_job
     # Check output for the run and decide if we should run this stage
     return if started? && !failed? # job has been started successfully
-    self.job_command = send(job_command_func)
-    self.command_stdout, self.command_stderr, status = Open3.capture3(job_command)
+    if step_function?
```

Comment:
The `step_function` attribute is part of the `PipelineRun` though, not the `PipelineRunStage`. I don't think this will work.

Reply:
Indeed! :)